plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageClientHttpTransportConfig.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage.config;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.units.Duration;
+import io.airlift.units.MaxDuration;
+import io.airlift.units.MinDuration;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+public class OpenLineageClientHttpTransportConfig
+{
+    private String url;
+    private String endpoint;
+    private Optional<String> apiKey = Optional.empty();
+    private Duration timeout = new Duration(5000, TimeUnit.MILLISECONDS);
+    private Map<String, String> headers = new HashMap<>();
+    private Map<String, String> urlParams = new HashMap<>();
+
+    @NotNull
+    public String getUrl()
+    {
+        return url;
+    }
+
+    @Config("openlineage-event-listener.transport.url")
+    @ConfigDescription("URL of receiving server. Explicitly set the scheme https:// to use symmetric encryption")
+    public OpenLineageClientHttpTransportConfig setUrl(String url)
+    {
+        this.url = url;
+        return this;
+    }
+
+    public String getEndpoint()
+    {
+        return endpoint;
+    }
+
+    @Config("openlineage-event-listener.transport.endpoint")
+    @ConfigDescription("Custom path for API receiving the events.")
+    public OpenLineageClientHttpTransportConfig setEndpoint(String endpoint)
+    {
+        this.endpoint = endpoint;
+        return this;
```

Comment:
Can we combine the url and endpoint config ?

Reply:
I configured them separately as this is the approach in OpenLineage client used in different integrations. Aditionally, having them separate unburdens users from having to know what is the default endpoint for OL lineage API.